packages/datadog-plugin-amqp10/src/producer.js

```diff
@@ -12,20 +13,22 @@ class Amqp10ProducerPlugin extends ProducerPlugin {
     const address = getAddress(link)
     const target = getShortName(link)
 
+    const hostDetails = resolveHostDetails(address.host)
+
     this.startSpan('amqp.send', {
       service: this.config.service || `${this.tracer._service}-amqp`,
       resource: ['send', target].filter(v => v).join(' '),
       kind: 'producer',
       meta: {
         'amqp.link.target.address': target,
         'amqp.link.role': 'sender',
-        'out.host': address.host,
-        'out.port': address.port,
         'amqp.link.name': link.name,
         'amqp.link.handle': link.handle,
         'amqp.connection.host': address.host,
         'amqp.connection.port': address.port,
-        'amqp.connection.user': address.user
+        'amqp.connection.user': address.user,
+        'network.destination.port': address.port,
+        ...hostDetails
```

Comment:
It looks like the order of the `...hostDetails` spread operation changes in each scenario, sometimes first, sometimes last. I believe later declared properties clobber earlier properties. Is the order intentional? If not it might be cleaner to always put them at the end or beginning.